commands/deploy_command.go

```diff
@@ -379,19 +408,41 @@ func (c *DeployCommand) uploadFromUrl(url string, mtaClient mtaclient.MtaClientO
 			totalBytesProcessed = jobResult.BytesProcessed
 		}
 
-		if len(mtaId) == 0 {
+		if len(jobResult.MtaId) == 0 {
 			select {
 			case <-timeout.C:
 				ui.Failed("Upload from URL timed out after 1 hour")
-				return "", "", true
+				return UploadFromUrlStatus{
+					FileId:          "",
+					MtaId:           "",
+					ClientActions:   make([]string, 0),
+					ExecutionStatus: Failure,
+				}
 			case <-ticker.C:
 			}
 		}
 	}
 	if progressBar != nil && totalBytesProcessed < progressBar.Total {
 		progressBar.Add64(progressBar.Total - totalBytesProcessed)
 	}
-	return file.ID, mtaId, false
+	return UploadFromUrlStatus{
+		FileId:          file.ID,
+		MtaId:           jobResult.MtaId,
+		ClientActions:   jobResult.ClientActions,
+		ExecutionStatus: Success,
+	}
+}
+
+func shouldRetryUpload(uploadFromUrlStatus UploadFromUrlStatus) bool {
+	if uploadFromUrlStatus.ExecutionStatus == Success {
+		return false
+	}
+	for _, clientAction := range uploadFromUrlStatus.ClientActions {
+		if clientAction == "RETRY_UPLOAD" {
+			return true
+		}
+	}
```

Comment:
Do we have similar retry mechanism when upload file to the backend of deploy-service?

Reply:
No, there's a single request/response, which will be retried by the rest client retry mechanism